### PR TITLE
Cross build for Scala 2.13: Putting the finishing touches

### PR DESCRIPTION
**Pull request by danicheg**
Hey there! It's yet another attempt to get closer to releasing the Scala 2.13 support for this great tool. This PR is built on top of the https://github.com/scalacenter/scalac-profiling/pull/32.

### What this PR does/aims:
1. Restores the work done by @tanishiking — I didn't even manage to start SBT :sad_pepe:
2. Adds support for the following Scala versions `2.12.16`, `2.12.17`, `2.12.18`, `2.13.9`, `2.13.10`, `2.13.11`, `2.13.12`.
3. Migrates the SBT plugin to `1.x`. Support for the `0.13` version was expunged — it's been 5+ years since the SBT `1.0` release 🎉  
4. Fixes the integration tests. Honestly, I had to prune most of the underlying projects (but add a new one). I am almost confident that the problem lies within the SBT plane, not the `scalac-profiling` itself. I faced weird errors like this one — some projects use `sbt-git` from `com.typesafe` artefacts but `scalac-profiling` uses ones from `com.github`, which leads to this `Some keys were defined with the same name but different types: 'gitReader' (com.github.sbt.git.ReadableGit, com.typesafe.sbt.git.ReadableGit), 'gitRunner' (sbt.Task[com.github.sbt.git.GitRunner], sbt.Task[com.typesafe.sbt.git.GitRunner])`. The key is that we still have some foreign projects plugged into the integration tests. And one could do follow-up work on bringing more of them. In short, I don't think we have to backport support of all of them. And it's neither a blocker for moving on.
5. Sets up the release plugin in a way it's done in other Scala Center projects. Probably, the settings of this git repo have to be reviewed on the eligibility of the `sbt-ci-release` plugin.
6. Tweaks some tiny things here and there.

### What's out of the scope of this PR:
1. Adding support for the `vscode`. If one argues that it is mandatory to be done to get this work released, I will give it a try. Or maybe someone from the Scala Center/community could do that.

### CODE REVIEW

**Comment by danicheg** [comment]
Tagging people from the original PR for better visibility @tanishiking @sjrd @SethTisue 🙏🏻

**Comment by danicheg** [comment]
Hey @tanishiking @sjrd! May I ask you to check this repo's settings for enabling the `Read and write permission` to the GA workflow? I predict this should fix the CI.
```
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

**Comment by sjrd** [comment]
Secrets are never provided to workflows coming from pull requests, for security reasons.

Use a `git` or `https` protocol to clone external git repositories, instead of `ssh`.

**Comment by danicheg** [comment]
@sjrd thank you for the advice. Now we have a "green CI". Hopefully, this is ready for the review.

**Comment by SethTisue** [comment]
Maybe one solution here would just be to merge it, roll a milestone or RC release, and invite people to test it?

**Comment by danicheg** [comment]
Most of the changes have been reviewed already in [another PR](https://github.com/scalacenter/scalac-profiling/pull/32). But in any case, it's better to review this before merging.
But publishing just a milestone would be the right approach in moving further :+1:

File: plugin/src/main/scala/ch/epfl/scala/profilers/ProfilingImpl.scala
Comment (sjrd) [inline plugin/src/main/scala/ch/epfl/scala/profilers/ProfilingImpl.scala:75]:
```suggestion
    kvs.groupBy(_._1.source).map {
      case (sf, posInfos: Map[Position, V]) => sf -> posInfos.valuesIterator.fold(empty)(aggregate)
    }
```
Taking a `view` to only perform one operation before converting back with `.toMap` is not useful.

File: plugin/src/main/scala/ch/epfl/scala/profilers/ProfilingImpl.scala
Comment (sjrd) [inline plugin/src/main/scala/ch/epfl/scala/profilers/ProfilingImpl.scala:86]:
```suggestion
      .map {
        case (sf, mi) => sf -> mi.copy(expansionNanos = toMillis(mi.expansionNanos))
      }
```

File: plugin/src/main/scala/ch/epfl/scala/profilers/ProfilingImpl.scala
Comment (sjrd) [inline plugin/src/main/scala/ch/epfl/scala/profilers/ProfilingImpl.scala:99]:
```suggestion
      .map {
        case (pos, mi) => pos -> mi.copy(expansionNanos = toMillis(mi.expansionNanos))
      }
```

File: plugin/src/main/scala/ch/epfl/scala/profilers/ProfilingImpl.scala
Comment (sjrd) [inline plugin/src/main/scala/ch/epfl/scala/profilers/ProfilingImpl.scala:123]:
```suggestion
    val perCallSite = implicitSearchesByPos.map {
      case (pos, i) => pos -> ImplicitInfo.apply(i)
    }
```

File: plugin/src/main/scala/ch/epfl/scala/profilers/ProfilingImpl.scala
Comment (sjrd) [inline plugin/src/main/scala/ch/epfl/scala/profilers/ProfilingImpl.scala:127]:
```suggestion
    val perType = implicitSearchesByType.map {
      case (pos, i) => pos -> ImplicitInfo.apply(i)
    }
```

File: build.sbt
Comment (sjrd) [inline build.sbt:223]:
```suggestion
   .settings(skip / publish := true)
```

**Review by sjrd — CHANGES_REQUESTED** [review]
I have some comments, but overall it looks like this is acceptable.

In addition to the comments below: the message of the last commit is not consistent with its content.

**Review by sjrd — APPROVED** [review]

**Comment by sjrd** [comment]
The release script apparently successfully published version `0-0.0.0+424-33b59cb6-SNAPSHOT`. You might want to give it a try.

**Comment by danicheg** [comment]
At least it publishes artefacts, hooray! Also, I'd like to bring the `baseVersion` to fix snapshot versioning. After fixing, it's going to be `1.0.0-711bf78-SNAPSHOT`. @sjrd wdyt?

**Comment by sjrd** [comment]
All I had to do for that was to actually back-publish a v1.0.0 tag corresponding (as best I could tell) to the actually release 1.0.0. I reran the publishing job and it now published as `1.0.0+41-33b59cb6-SNAPSHOT`.

**Comment by danicheg** [comment]
So, we have to wait for the artefacts to appear in Maven Central. I will return with some results of testing in projects which are near at hand. Although, I tested the plugin by publishing it locally.

**Comment by danicheg** [comment]
Ah, yes. Perhaps I encountered some flaky on the first try. As of now, adding the snapshot resolver to the `plugins.sbt` (`resolvers += Resolver.sonatypeRepo("snapshots")`) makes it work well.

**Comment by sjrd** [comment]
Good. Thanks for checking. Shall I release it as a true version, then?

**Comment by danicheg** [comment]
Also, I managed to perform e2e testing on a Scala 2.13 project, everything appears to be in order. 

> Shall I release it as a true version, then?

As we agreed with @SethTisue, it should be fine to publish a milestone release (or an RC) and ask the community for testing if they are interested. Does it work for you? On the opposite, I can draft a small how-to with the snapshot version, and we can wait for the feedback from the community.

**Comment by sjrd** [comment]
1.1.0-RC1 is on its way to Maven Central.

**Comment by jtjeferreira** [comment]
I gave 1.1.0-RC1 a try and besides https://github.com/scalacenter/scalac-profiling/pull/34 it worked fine

**Pull request merged**
